### PR TITLE
Update peer pool documentation after implementation

### PR DESCRIPTION
**Pull request by sylane**
Update the documentation to reflect the implementation from https://github.com/aeternity/epoch/pull/1252 .

### CODE REVIEW

File: GOSSIP.md [hunk 1]
```diff
@@ -130,32 +130,41 @@ nodes to join the cluster even if the trusted nodes inbound limit has been
 reached. In addition, this prevent a node, in particular a trusted node, to
 have so much inbound connections that it cannot reach its number of outbound
 connections; a guaranteed number of outbound connection is crucial for proper
-propagation of new blocks.
+propagation of new blocks. For all nodes to reach there established number of
+outbound connections, the soft maximum limit for inbound connections must be
+smaller than the maximum number of nodes minus the wanted number of outgoing
```
Comment (knarz) [inline GOSSIP.md]: The maximum number of open connections instead of »nodes«, I assume?

File: GOSSIP.md [hunk 2]
```diff
@@ -130,32 +130,41 @@ nodes to join the cluster even if the trusted nodes inbound limit has been
...
+connections.
 
-When started, the node will iteratively pick a random peer from the verified
-pool (not yet connected), that is from a different group
-([See Peer Groups](#peer-groups)) than any actual outbound connection.
-In case there are no more peers available in the verified pool, a random one is
-picked from the unverified pool. If the connection succeeds, an
+The node will first burst-connect to the trusted peers and then periodically
+connects to more peers until it reaches the maximum number of outbound
+connections; the delay between connection should be small enough so the nodes
```
Comment (knarz) [inline GOSSIP.md]: Delay between connections to new peers?

File: GOSSIP.md [hunk 3]
```diff
@@ -257,26 +265,27 @@ are selected from the secret and other discriminators.
...
+- Probability of adding a Nth duplicated reference of an existing peer to the
+unverified pool: `1/2^N`
 - Period of verified peer random peer check when the maximum number
-of verified connections has been reached: 60 seconds
+of verified connections has been reached: `60 seconds`
 - Period of new peer connection up to the maximum number of verified
-connections: 20 seconds.
-- Gossip ping frequency: 120 seconds
-- Maximum time to get a ping from an inbound connection: 30 seconds
+connections: `min(30, 2^OUTBOUND_CONNECTION_COUNT) seconds`.
```
Comment (sylane) [inline GOSSIP.md]: I Forgot to update this it is `min(30, 2^(OUTBOUND_CONNECTION_COUNT - 1))`. I'll change it next week.

File: GOSSIP.md [hunk 4]
```diff
@@ -130,32 +130,41 @@ nodes to join the cluster even if the trusted nodes inbound limit has been
...
+connects to more peers until it reaches the maximum number of outbound
+connections; the delay between connection should be small enough so the nodes
+has enough outbound connections to work properly, but large enough so it has
+enough peers to choose from. This is done by having a variable delay between
+connections of `2^(OUTBOUND_CONNECTION_COUNT - 1)` seconds when
+`OUTBOUND_CONNECTION_COUNT` is greater than `0` with a maximum of `30` seconds;
+this allow the node to reach 5 outbound connection under `15` seconds and
+connect to the last peer when having received around `20` gossip messages.
+
+The node iteratively pick a random peer (not yet connected) from either
```
Comment (knarz) [inline GOSSIP.md]: s/pick/picks/

File: GOSSIP.md [hunk 5]
```diff
@@ -130,32 +130,41 @@ nodes to join the cluster even if the trusted nodes inbound limit has been
...
 a ping message, only establishing the Noise connection. This ensure the pool
 contains enough reachable peers to reduce the chance of a Sybil attack were
 most of the good peers are unreachable augmenting the probability of only
-hostile node to get selected.
+hostile node getting selected. (Not yet implemented).
 
-If a peer changes its address but not its key, it will be removed after
-the normal retry policy is exausted. Then the new address will be added through
-the usual gossip exchanges.
+If a peer changes its address but not its key, the new address will **never**
```
Comment (knarz) [inline GOSSIP.md]: Address meaning IP address, I presume?

File: GOSSIP.md [hunk 6]
```diff
@@ -130,32 +130,41 @@ nodes to join the cluster even if the trusted nodes inbound limit has been
...
+be updated through gossip. This is to prevent an hostil node from gossiping
+bad address for known good nodes and making them unreachable. The peer will be
```
Comment (knarz) [inline GOSSIP.md]: s/address/addresses/

File: GOSSIP.md (hunk 1 above)
Comment (sylane) [reply GOSSIP.md]: @knarz I actually mean `total number of nodes` in the cluster. If there are fewer nodes than the maximum number of outbound connection, the trusted nodes will have all the other nodes in the cluster as inbound connections and as such will never reach there the maximum number of outbound connections.

**Pull request merged**